GCC 13 build failure

### Steps to reproduce

Build with GCC 13.0.1.

### Expected behaviour

Successful build.

### Actual behaviour

Build error:
```
[706/1264] /usr/bin/g++ -DCRL_USE_COMMON_QUEUE -DDESKTOP_APP_DISABLE_CRASH_REPORTS -DDESKTOP_APP_USE_PACKAGED -DG_LOG_DOMAIN=\"Telegram\" -DKCOREADDONS_EXPORT="" -DLOTTIE_USE_CACHE -DLOT_BUILD -DQT_CORE5COMPAT_LIB -DQT_CORE_LIB -DQT_DBUS_LIB -DQT_GUI_LIB -DQT_IMPLICIT_QCHAR_CONSTRUCTION -DQT_NETWORK_LIB -DQT_NO_CAST_FROM_BYTEARRAY -DQT_NO_DEBUG -DQT_NO_KEYWORDS -DQT_OPENGLWIDGETS_LIB -DQT_OPENGL_LIB -DQT_QMLINTEGRATION_LIB -DQT_QMLMODELS_LIB -DQT_QML_LIB -DQT_QUICKWIDGETS_LIB -DQT_QUICK_LIB -DQT_SVG_LIB -DQT_WAYLANDCOMPOSITOR_LIB -DQT_WIDGETS_LIB -DTDESKTOP_API_HASH=d524b414d21f4d37f08684c1df41ac9c -DTDESKTOP_API_ID=611335 -DTDESKTOP_DISABLE_AUTOUPDATE -DTGCALLS_USE_STD_OPTIONAL -I/builddir/build/BUILD/tdesktop-4.5.3-full/redhat-linux-build/Telegram/Telegram_autogen/include -I/builddir/build/BUILD/tdesktop-4.5.3-full/redhat-linux-build/Telegram -I/builddir/build/BUILD/tdesktop-4.5.3-full/Telegram/SourceFiles -I/builddir/build/BUILD/tdesktop-4.5.3-full/redhat-linux-build/Telegram/gen -I/builddir/build/BUILD/tdesktop-4.5.3-full/Telegram/lib_spellcheck -I/builddir/build/BUILD/tdesktop-4.5.3-full/Telegram/ThirdParty/tgcalls -I/builddir/build/BUILD/tdesktop-4.5.3-full/Telegram/ThirdParty/libtgvoip -I/builddir/build/BUILD/tdesktop-4.5.3-full/Telegram/lib_base -I/builddir/build/BUILD/tdesktop-4.5.3-full/Telegram/lib_rpl -I/builddir/build/BUILD/tdesktop-4.5.3-full/Telegram/lib_crl -I/builddir/build/BUILD/tdesktop-4.5.3-full/Telegram/lib_tl -I/builddir/build/BUILD/tdesktop-4.5.3-full/redhat-linux-build/Telegram/lib_ui/gen -I/builddir/build/BUILD/tdesktop-4.5.3-full/Telegram/lib_ui -I/builddir/build/BUILD/tdesktop-4.5.3-full/Telegram/lib_ui/emoji_suggestions -I/builddir/build/BUILD/tdesktop-4.5.3-full/Telegram/lib_lottie -I/builddir/build/BUILD/tdesktop-4.5.3-full/Telegram/lib_webrtc -I/builddir/build/BUILD/tdesktop-4.5.3-full/Telegram/lib_storage -I/builddir/build/BUILD/tdesktop-4.5.3-full/Telegram/lib_qr -I/builddir/build/BUILD/tdesktop-4.5.3-full/Telegram/lib_webview -I/builddir/build/BUILD/tdesktop-4.5.3-full/Telegram/SourceFiles/payments -I/builddir/build/BUILD/tdesktop-4.5.3-full/Telegram/ThirdParty/rlottie/inc -isystem /usr/include/minizip -isystem /usr/include/qt6/QtCore -isystem /usr/include/qt6 -isystem /usr/lib64/qt6/mkspecs/linux-g++ -isystem /usr/include/qt6/QtCore/6.4.2 -isystem /usr/include/qt6/QtCore/6.4.2/QtCore -isystem /usr/include/qt6/QtCore5Compat -isystem /usr/include/qt6/QtGui -isystem /usr/include/qt6/QtGui/6.4.2 -isystem /usr/include/qt6/QtGui/6.4.2/QtGui -isystem /usr/include/qt6/QtOpenGL -isystem /usr/include/qt6/QtWidgets -isystem /usr/include/qt6/QtWidgets/6.4.2 -isystem /usr/include/qt6/QtWidgets/6.4.2/QtWidgets -isystem /usr/include/qt6/QtOpenGLWidgets -isystem /usr/include/qt6/QtNetwork -isystem /usr/include/qt6/QtSvg -isystem /usr/include/qt6/QtDBus -isystem /usr/include/qt6/QtQml -isystem /usr/include/qt6/QtQmlIntegration -isystem /usr/include/qt6/QtQuick -isystem /usr/include/qt6/QtQmlModels -isystem /usr/include/qt6/QtQuickWidgets -isystem /usr/include/qt6/QtWaylandCompositor -isystem /usr/share -isystem /builddir/build/BUILD/tdesktop-4.5.3-full/Telegram/ThirdParty/kcoreaddons/src/lib/io -isystem /builddir/build/BUILD/tdesktop-4.5.3-full/Telegram/ThirdParty/kcoreaddons/src/lib/util -isystem /builddir/build/BUILD/tdesktop-4.5.3-full/cmake/external/kcoreaddons/headers/public -isystem /usr/include/giomm-2.68 -isystem /usr/lib64/giomm-2.68/include -isystem /usr/include/glib-2.0 -isystem /usr/lib64/glib-2.0/include -isystem /usr/include/sysprof-4 -isystem /usr/include/libmount -isystem /usr/include/blkid -isystem /usr/include/glibmm-2.68 -isystem /usr/lib64/glibmm-2.68/include -isystem /usr/include/sigc++-3.0 -isystem /usr/lib64/sigc++-3.0/include -isystem /usr/include/gio-unix-2.0 -isystem /usr/include/compat-ffmpeg4 -isystem /usr/include/AL -isystem /usr/include/qrcodegencpp -O2 -flto=auto -ffat-lto-objects -fexceptions -g1 -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -DNDEBUG -fPIC -fno-strict-aliasing -pipe -Wall -Wextra -Wno-unused-parameter -Wno-switch -Wno-maybe-uninitialized -Wno-missing-field-initializers -Wno-sign-compare -Wno-deprecated -Wno-deprecated-declarations -pthread -std=gnu++20 -Winvalid-pch -include /builddir/build/BUILD/tdesktop-4.5.3-full/redhat-linux-build/Telegram/CMakeFiles/Telegram.dir/cmake_pch.hxx -MD -MT Telegram/CMakeFiles/Telegram.dir/SourceFiles/boxes/peers/edit_forum_topic_box.cpp.o -MF Telegram/CMakeFiles/Telegram.dir/SourceFiles/boxes/peers/edit_forum_topic_box.cpp.o.d -o Telegram/CMakeFiles/Telegram.dir/SourceFiles/boxes/peers/edit_forum_topic_box.cpp.o -c /builddir/build/BUILD/tdesktop-4.5.3-full/Telegram/SourceFiles/boxes/peers/edit_forum_topic_box.cpp
FAILED: Telegram/CMakeFiles/Telegram.dir/SourceFiles/boxes/peers/edit_forum_topic_box.cpp.o 
/usr/bin/g++ -DCRL_USE_COMMON_QUEUE -DDESKTOP_APP_DISABLE_CRASH_REPORTS -DDESKTOP_APP_USE_PACKAGED -DG_LOG_DOMAIN=\"Telegram\" -DKCOREADDONS_EXPORT="" -DLOTTIE_USE_CACHE -DLOT_BUILD -DQT_CORE5COMPAT_LIB -DQT_CORE_LIB -DQT_DBUS_LIB -DQT_GUI_LIB -DQT_IMPLICIT_QCHAR_CONSTRUCTION -DQT_NETWORK_LIB -DQT_NO_CAST_FROM_BYTEARRAY -DQT_NO_DEBUG -DQT_NO_KEYWORDS -DQT_OPENGLWIDGETS_LIB -DQT_OPENGL_LIB -DQT_QMLINTEGRATION_LIB -DQT_QMLMODELS_LIB -DQT_QML_LIB -DQT_QUICKWIDGETS_LIB -DQT_QUICK_LIB -DQT_SVG_LIB -DQT_WAYLANDCOMPOSITOR_LIB -DQT_WIDGETS_LIB -DTDESKTOP_API_HASH=d524b414d21f4d37f08684c1df41ac9c -DTDESKTOP_API_ID=611335 -DTDESKTOP_DISABLE_AUTOUPDATE -DTGCALLS_USE_STD_OPTIONAL -I/builddir/build/BUILD/tdesktop-4.5.3-full/redhat-linux-build/Telegram/Telegram_autogen/include -I/builddir/build/BUILD/tdesktop-4.5.3-full/redhat-linux-build/Telegram -I/builddir/build/BUILD/tdesktop-4.5.3-full/Telegram/SourceFiles -I/builddir/build/BUILD/tdesktop-4.5.3-full/redhat-linux-build/Telegram/gen -I/builddir/build/BUILD/tdesktop-4.5.3-full/Telegram/lib_spellcheck -I/builddir/build/BUILD/tdesktop-4.5.3-full/Telegram/ThirdParty/tgcalls -I/builddir/build/BUILD/tdesktop-4.5.3-full/Telegram/ThirdParty/libtgvoip -I/builddir/build/BUILD/tdesktop-4.5.3-full/Telegram/lib_base -I/builddir/build/BUILD/tdesktop-4.5.3-full/Telegram/lib_rpl -I/builddir/build/BUILD/tdesktop-4.5.3-full/Telegram/lib_crl -I/builddir/build/BUILD/tdesktop-4.5.3-full/Telegram/lib_tl -I/builddir/build/BUILD/tdesktop-4.5.3-full/redhat-linux-build/Telegram/lib_ui/gen -I/builddir/build/BUILD/tdesktop-4.5.3-full/Telegram/lib_ui -I/builddir/build/BUILD/tdesktop-4.5.3-full/Telegram/lib_ui/emoji_suggestions -I/builddir/build/BUILD/tdesktop-4.5.3-full/Telegram/lib_lottie -I/builddir/build/BUILD/tdesktop-4.5.3-full/Telegram/lib_webrtc -I/builddir/build/BUILD/tdesktop-4.5.3-full/Telegram/lib_storage -I/builddir/build/BUILD/tdesktop-4.5.3-full/Telegram/lib_qr -I/builddir/build/BUILD/tdesktop-4.5.3-full/Telegram/lib_webview -I/builddir/build/BUILD/tdesktop-4.5.3-full/Telegram/SourceFiles/payments -I/builddir/build/BUILD/tdesktop-4.5.3-full/Telegram/ThirdParty/rlottie/inc -isystem /usr/include/minizip -isystem /usr/include/qt6/QtCore -isystem /usr/include/qt6 -isystem /usr/lib64/qt6/mkspecs/linux-g++ -isystem /usr/include/qt6/QtCore/6.4.2 -isystem /usr/include/qt6/QtCore/6.4.2/QtCore -isystem /usr/include/qt6/QtCore5Compat -isystem /usr/include/qt6/QtGui -isystem /usr/include/qt6/QtGui/6.4.2 -isystem /usr/include/qt6/QtGui/6.4.2/QtGui -isystem /usr/include/qt6/QtOpenGL -isystem /usr/include/qt6/QtWidgets -isystem /usr/include/qt6/QtWidgets/6.4.2 -isystem /usr/include/qt6/QtWidgets/6.4.2/QtWidgets -isystem /usr/include/qt6/QtOpenGLWidgets -isystem /usr/include/qt6/QtNetwork -isystem /usr/include/qt6/QtSvg -isystem /usr/include/qt6/QtDBus -isystem /usr/include/qt6/QtQml -isystem /usr/include/qt6/QtQmlIntegration -isystem /usr/include/qt6/QtQuick -isystem /usr/include/qt6/QtQmlModels -isystem /usr/include/qt6/QtQuickWidgets -isystem /usr/include/qt6/QtWaylandCompositor -isystem /usr/share -isystem /builddir/build/BUILD/tdesktop-4.5.3-full/Telegram/ThirdParty/kcoreaddons/src/lib/io -isystem /builddir/build/BUILD/tdesktop-4.5.3-full/Telegram/ThirdParty/kcoreaddons/src/lib/util -isystem /builddir/build/BUILD/tdesktop-4.5.3-full/cmake/external/kcoreaddons/headers/public -isystem /usr/include/giomm-2.68 -isystem /usr/lib64/giomm-2.68/include -isystem /usr/include/glib-2.0 -isystem /usr/lib64/glib-2.0/include -isystem /usr/include/sysprof-4 -isystem /usr/include/libmount -isystem /usr/include/blkid -isystem /usr/include/glibmm-2.68 -isystem /usr/lib64/glibmm-2.68/include -isystem /usr/include/sigc++-3.0 -isystem /usr/lib64/sigc++-3.0/include -isystem /usr/include/gio-unix-2.0 -isystem /usr/include/compat-ffmpeg4 -isystem /usr/include/AL -isystem /usr/include/qrcodegencpp -O2 -flto=auto -ffat-lto-objects -fexceptions -g1 -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -DNDEBUG -fPIC -fno-strict-aliasing -pipe -Wall -Wextra -Wno-unused-parameter -Wno-switch -Wno-maybe-uninitialized -Wno-missing-field-initializers -Wno-sign-compare -Wno-deprecated -Wno-deprecated-declarations -pthread -std=gnu++20 -Winvalid-pch -include /builddir/build/BUILD/tdesktop-4.5.3-full/redhat-linux-build/Telegram/CMakeFiles/Telegram.dir/cmake_pch.hxx -MD -MT Telegram/CMakeFiles/Telegram.dir/SourceFiles/boxes/peers/edit_forum_topic_box.cpp.o -MF Telegram/CMakeFiles/Telegram.dir/SourceFiles/boxes/peers/edit_forum_topic_box.cpp.o.d -o Telegram/CMakeFiles/Telegram.dir/SourceFiles/boxes/peers/edit_forum_topic_box.cpp.o -c /builddir/build/BUILD/tdesktop-4.5.3-full/Telegram/SourceFiles/boxes/peers/edit_forum_topic_box.cpp
In file included from /builddir/build/BUILD/tdesktop-4.5.3-full/Telegram/SourceFiles/history/view/history_view_replies_section.h:13,
                 from /builddir/build/BUILD/tdesktop-4.5.3-full/Telegram/SourceFiles/boxes/peers/edit_forum_topic_box.cpp:29:
/builddir/build/BUILD/tdesktop-4.5.3-full/Telegram/SourceFiles/history/view/history_view_list_widget.h:405:15: error: declaration of 'using HistoryView::ListWidget::PointState = enum class HistoryView::PointState' changes meaning of 'PointState' [-fpermissive]
  405 |         using PointState = HistoryView::PointState;
      |               ^~~~~~~~~~
/builddir/build/BUILD/tdesktop-4.5.3-full/Telegram/SourceFiles/history/view/history_view_list_widget.h:365:17: note: used here to mean 'enum class HistoryView::PointState'
  365 |                 PointState pointState;
      |                 ^~~~~~~~~~
In file included from /builddir/build/BUILD/tdesktop-4.5.3-full/Telegram/SourceFiles/history/view/history_view_list_widget.h:17:
/builddir/build/BUILD/tdesktop-4.5.3-full/Telegram/SourceFiles/history/view/history_view_element.h:47:25: note: declared here
   47 | enum class PointState : char;
      |                         ^~~~
```

### Operating system

Fedora 38

### Version of Telegram Desktop

4.5.3

### Installation source

Other (unofficial) source

### Logs

```shell
Full build log: https://koji.rpmfusion.org/kojifiles/work/tasks/1987/581987/build.log
```

> you can also try such workaround:

It works. New error:

```
[1172/1282] /usr/bin/g++ -DCRL_USE_COMMON_QUEUE -DDESKTOP_APP_DISABLE_CRASH_REPORTS -DDESKTOP_APP_USE_PACKAGED -DG_LOG_DOMAIN=\"Telegram\" -DKCOREADDONS_EXPORT="" -DLOTTIE_USE_CACHE -DLOT_BUILD -DQT_CORE5COMPAT_LIB -DQT_CORE_LIB -DQT_DBUS_LIB -DQT_GUI_LIB -DQT_IMPLICIT_QCHAR_CONSTRUCTION -DQT_NETWORK_LIB -DQT_NO_CAST_FROM_BYTEARRAY -DQT_NO_DEBUG -DQT_NO_KEYWORDS -DQT_OPENGLWIDGETS_LIB -DQT_OPENGL_LIB -DQT_QMLINTEGRATION_LIB -DQT_QMLMODELS_LIB -DQT_QML_LIB -DQT_QUICKWIDGETS_LIB -DQT_QUICK_LIB -DQT_SVG_LIB -DQT_WAYLANDCOMPOSITOR_LIB -DQT_WIDGETS_LIB -DTDESKTOP_API_HASH=d524b414d21f4d37f08684c1df41ac9c -DTDESKTOP_API_ID=611335 -DTDESKTOP_DISABLE_AUTOUPDATE -DTGCALLS_USE_STD_OPTIONAL -I/builddir/build/BUILD/tdesktop-4.6.0-full/redhat-linux-build/Telegram/Telegram_autogen/include -I/builddir/build/BUILD/tdesktop-4.6.0-full/redhat-linux-build/Telegram -I/builddir/build/BUILD/tdesktop-4.6.0-full/Telegram/SourceFiles -I/builddir/build/BUILD/tdesktop-4.6.0-full/redhat-linux-build/Telegram/gen -I/builddir/build/BUILD/tdesktop-4.6.0-full/Telegram/ThirdParty/tgcalls -I/builddir/build/BUILD/tdesktop-4.6.0-full/Telegram/ThirdParty/libtgvoip -I/builddir/build/BUILD/tdesktop-4.6.0-full/Telegram/lib_base -I/builddir/build/BUILD/tdesktop-4.6.0-full/Telegram/lib_rpl -I/builddir/build/BUILD/tdesktop-4.6.0-full/Telegram/lib_crl -I/builddir/build/BUILD/tdesktop-4.6.0-full/Telegram/lib_tl -I/builddir/build/BUILD/tdesktop-4.6.0-full/redhat-linux-build/Telegram/lib_ui/gen -I/builddir/build/BUILD/tdesktop-4.6.0-full/Telegram/lib_ui -I/builddir/build/BUILD/tdesktop-4.6.0-full/Telegram/lib_ui/emoji_suggestions -I/builddir/build/BUILD/tdesktop-4.6.0-full/Telegram/lib_lottie -I/builddir/build/BUILD/tdesktop-4.6.0-full/Telegram/lib_webrtc -I/builddir/build/BUILD/tdesktop-4.6.0-full/Telegram/lib_spellcheck -I/builddir/build/BUILD/tdesktop-4.6.0-full/Telegram/lib_storage -I/builddir/build/BUILD/tdesktop-4.6.0-full/Telegram/lib_qr -I/builddir/build/BUILD/tdesktop-4.6.0-full/Telegram/lib_webview -I/builddir/build/BUILD/tdesktop-4.6.0-full/Telegram/SourceFiles/payments -I/builddir/build/BUILD/tdesktop-4.6.0-full/Telegram/ThirdParty/rlottie/inc -isystem /usr/include/minizip -isystem /usr/include/qt6/QtCore -isystem /usr/include/qt6 -isystem /usr/lib64/qt6/mkspecs/linux-g++ -isystem /usr/include/qt6/QtCore/6.4.2 -isystem /usr/include/qt6/QtCore/6.4.2/QtCore -isystem /usr/include/qt6/QtCore5Compat -isystem /usr/include/qt6/QtGui -isystem /usr/include/qt6/QtGui/6.4.2 -isystem /usr/include/qt6/QtGui/6.4.2/QtGui -isystem /usr/include/qt6/QtOpenGL -isystem /usr/include/qt6/QtWidgets -isystem /usr/include/qt6/QtWidgets/6.4.2 -isystem /usr/include/qt6/QtWidgets/6.4.2/QtWidgets -isystem /usr/include/qt6/QtOpenGLWidgets -isystem /usr/include/qt6/QtNetwork -isystem /usr/include/qt6/QtSvg -isystem /usr/include/qt6/QtDBus -isystem /usr/include/qt6/QtQml -isystem /usr/include/qt6/QtQmlIntegration -isystem /usr/include/qt6/QtQuick -isystem /usr/include/qt6/QtQmlModels -isystem /usr/include/qt6/QtQuickWidgets -isystem /usr/include/qt6/QtWaylandCompositor -isystem /usr/share -isystem /builddir/build/BUILD/tdesktop-4.6.0-full/Telegram/ThirdParty/kcoreaddons/src/lib/io -isystem /builddir/build/BUILD/tdesktop-4.6.0-full/Telegram/ThirdParty/kcoreaddons/src/lib/util -isystem /builddir/build/BUILD/tdesktop-4.6.0-full/cmake/external/kcoreaddons/headers/public -isystem /usr/include/giomm-2.68 -isystem /usr/lib64/giomm-2.68/include -isystem /usr/include/glib-2.0 -isystem /usr/lib64/glib-2.0/include -isystem /usr/include/sysprof-4 -isystem /usr/include/libmount -isystem /usr/include/blkid -isystem /usr/include/glibmm-2.68 -isystem /usr/lib64/glibmm-2.68/include -isystem /usr/include/sigc++-3.0 -isystem /usr/lib64/sigc++-3.0/include -isystem /usr/include/gio-unix-2.0 -isystem /usr/include/compat-ffmpeg4 -isystem /usr/include/AL -isystem /usr/include/qrcodegencpp -O2 -flto=auto -ffat-lto-objects -fexceptions -g1 -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -DNDEBUG -fPIC -fno-strict-aliasing -pipe -Wall -Wextra -Wno-unused-parameter -Wno-switch -Wno-maybe-uninitialized -Wno-missing-field-initializers -Wno-sign-compare -Wno-deprecated -Wno-deprecated-declarations -pthread -std=gnu++20 -Winvalid-pch -include /builddir/build/BUILD/tdesktop-4.6.0-full/redhat-linux-build/Telegram/CMakeFiles/Telegram.dir/cmake_pch.hxx -MD -MT Telegram/CMakeFiles/Telegram.dir/SourceFiles/settings/settings_advanced.cpp.o -MF Telegram/CMakeFiles/Telegram.dir/SourceFiles/settings/settings_advanced.cpp.o.d -o Telegram/CMakeFiles/Telegram.dir/SourceFiles/settings/settings_advanced.cpp.o -c /builddir/build/BUILD/tdesktop-4.6.0-full/Telegram/SourceFiles/settings/settings_advanced.cpp
FAILED: Telegram/CMakeFiles/Telegram.dir/SourceFiles/settings/settings_advanced.cpp.o 
/usr/bin/g++ -DCRL_USE_COMMON_QUEUE -DDESKTOP_APP_DISABLE_CRASH_REPORTS -DDESKTOP_APP_USE_PACKAGED -DG_LOG_DOMAIN=\"Telegram\" -DKCOREADDONS_EXPORT="" -DLOTTIE_USE_CACHE -DLOT_BUILD -DQT_CORE5COMPAT_LIB -DQT_CORE_LIB -DQT_DBUS_LIB -DQT_GUI_LIB -DQT_IMPLICIT_QCHAR_CONSTRUCTION -DQT_NETWORK_LIB -DQT_NO_CAST_FROM_BYTEARRAY -DQT_NO_DEBUG -DQT_NO_KEYWORDS -DQT_OPENGLWIDGETS_LIB -DQT_OPENGL_LIB -DQT_QMLINTEGRATION_LIB -DQT_QMLMODELS_LIB -DQT_QML_LIB -DQT_QUICKWIDGETS_LIB -DQT_QUICK_LIB -DQT_SVG_LIB -DQT_WAYLANDCOMPOSITOR_LIB -DQT_WIDGETS_LIB -DTDESKTOP_API_HASH=d524b414d21f4d37f08684c1df41ac9c -DTDESKTOP_API_ID=611335 -DTDESKTOP_DISABLE_AUTOUPDATE -DTGCALLS_USE_STD_OPTIONAL -I/builddir/build/BUILD/tdesktop-4.6.0-full/redhat-linux-build/Telegram/Telegram_autogen/include -I/builddir/build/BUILD/tdesktop-4.6.0-full/redhat-linux-build/Telegram -I/builddir/build/BUILD/tdesktop-4.6.0-full/Telegram/SourceFiles -I/builddir/build/BUILD/tdesktop-4.6.0-full/redhat-linux-build/Telegram/gen -I/builddir/build/BUILD/tdesktop-4.6.0-full/Telegram/ThirdParty/tgcalls -I/builddir/build/BUILD/tdesktop-4.6.0-full/Telegram/ThirdParty/libtgvoip -I/builddir/build/BUILD/tdesktop-4.6.0-full/Telegram/lib_base -I/builddir/build/BUILD/tdesktop-4.6.0-full/Telegram/lib_rpl -I/builddir/build/BUILD/tdesktop-4.6.0-full/Telegram/lib_crl -I/builddir/build/BUILD/tdesktop-4.6.0-full/Telegram/lib_tl -I/builddir/build/BUILD/tdesktop-4.6.0-full/redhat-linux-build/Telegram/lib_ui/gen -I/builddir/build/BUILD/tdesktop-4.6.0-full/Telegram/lib_ui -I/builddir/build/BUILD/tdesktop-4.6.0-full/Telegram/lib_ui/emoji_suggestions -I/builddir/build/BUILD/tdesktop-4.6.0-full/Telegram/lib_lottie -I/builddir/build/BUILD/tdesktop-4.6.0-full/Telegram/lib_webrtc -I/builddir/build/BUILD/tdesktop-4.6.0-full/Telegram/lib_spellcheck -I/builddir/build/BUILD/tdesktop-4.6.0-full/Telegram/lib_storage -I/builddir/build/BUILD/tdesktop-4.6.0-full/Telegram/lib_qr -I/builddir/build/BUILD/tdesktop-4.6.0-full/Telegram/lib_webview -I/builddir/build/BUILD/tdesktop-4.6.0-full/Telegram/SourceFiles/payments -I/builddir/build/BUILD/tdesktop-4.6.0-full/Telegram/ThirdParty/rlottie/inc -isystem /usr/include/minizip -isystem /usr/include/qt6/QtCore -isystem /usr/include/qt6 -isystem /usr/lib64/qt6/mkspecs/linux-g++ -isystem /usr/include/qt6/QtCore/6.4.2 -isystem /usr/include/qt6/QtCore/6.4.2/QtCore -isystem /usr/include/qt6/QtCore5Compat -isystem /usr/include/qt6/QtGui -isystem /usr/include/qt6/QtGui/6.4.2 -isystem /usr/include/qt6/QtGui/6.4.2/QtGui -isystem /usr/include/qt6/QtOpenGL -isystem /usr/include/qt6/QtWidgets -isystem /usr/include/qt6/QtWidgets/6.4.2 -isystem /usr/include/qt6/QtWidgets/6.4.2/QtWidgets -isystem /usr/include/qt6/QtOpenGLWidgets -isystem /usr/include/qt6/QtNetwork -isystem /usr/include/qt6/QtSvg -isystem /usr/include/qt6/QtDBus -isystem /usr/include/qt6/QtQml -isystem /usr/include/qt6/QtQmlIntegration -isystem /usr/include/qt6/QtQuick -isystem /usr/include/qt6/QtQmlModels -isystem /usr/include/qt6/QtQuickWidgets -isystem /usr/include/qt6/QtWaylandCompositor -isystem /usr/share -isystem /builddir/build/BUILD/tdesktop-4.6.0-full/Telegram/ThirdParty/kcoreaddons/src/lib/io -isystem /builddir/build/BUILD/tdesktop-4.6.0-full/Telegram/ThirdParty/kcoreaddons/src/lib/util -isystem /builddir/build/BUILD/tdesktop-4.6.0-full/cmake/external/kcoreaddons/headers/public -isystem /usr/include/giomm-2.68 -isystem /usr/lib64/giomm-2.68/include -isystem /usr/include/glib-2.0 -isystem /usr/lib64/glib-2.0/include -isystem /usr/include/sysprof-4 -isystem /usr/include/libmount -isystem /usr/include/blkid -isystem /usr/include/glibmm-2.68 -isystem /usr/lib64/glibmm-2.68/include -isystem /usr/include/sigc++-3.0 -isystem /usr/lib64/sigc++-3.0/include -isystem /usr/include/gio-unix-2.0 -isystem /usr/include/compat-ffmpeg4 -isystem /usr/include/AL -isystem /usr/include/qrcodegencpp -O2 -flto=auto -ffat-lto-objects -fexceptions -g1 -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -DNDEBUG -fPIC -fno-strict-aliasing -pipe -Wall -Wextra -Wno-unused-parameter -Wno-switch -Wno-maybe-uninitialized -Wno-missing-field-initializers -Wno-sign-compare -Wno-deprecated -Wno-deprecated-declarations -pthread -std=gnu++20 -Winvalid-pch -include /builddir/build/BUILD/tdesktop-4.6.0-full/redhat-linux-build/Telegram/CMakeFiles/Telegram.dir/cmake_pch.hxx -MD -MT Telegram/CMakeFiles/Telegram.dir/SourceFiles/settings/settings_advanced.cpp.o -MF Telegram/CMakeFiles/Telegram.dir/SourceFiles/settings/settings_advanced.cpp.o.d -o Telegram/CMakeFiles/Telegram.dir/SourceFiles/settings/settings_advanced.cpp.o -c /builddir/build/BUILD/tdesktop-4.6.0-full/Telegram/SourceFiles/settings/settings_advanced.cpp
In file included from /builddir/build/BUILD/tdesktop-4.6.0-full/Telegram/lib_rpl/rpl/rpl.h:28,
                 from /builddir/build/BUILD/tdesktop-4.6.0-full/Telegram/SourceFiles/stdafx.h:109,
                 from /builddir/build/BUILD/tdesktop-4.6.0-full/redhat-linux-build/Telegram/CMakeFiles/Telegram.dir/cmake_pch.hxx:5,
                 from <command-line>:
/builddir/build/BUILD/tdesktop-4.6.0-full/Telegram/lib_rpl/rpl/filter.h:32:14: error: mangling of 'auto rpl::details::filter_helper<Predicate>::operator()(rpl::producer<OtherValue, OtherError, OtherGenerator>&&) [with Value = bool; Error = rpl::no_error; Generator = rpl::details::type_erased_generator<bool, rpl::no_error>; <template-parameter-2-4> = void; Predicate = Settings::SetupSystemIntegrationContent(Window::SessionController*, gsl::not_null<Ui::VerticalLayout*>)::<lambda(bool)>]' as '_ZN3rpl7details13filter_helperIZN8Settings29SetupSystemIntegrationContentEPN6Window17SessionControllerEN3gsl8not_nullIPN2Ui14VerticalLayoutEEEEUlbE10_EclIbNS_8no_errorENS0_21type_erased_generatorIbSF_EEvEEDaONS_8producerIT_T0_T1_EE' conflicts with a previous mangle
   32 |         auto operator()(producer<Value, Error, Generator> &&initial) {
      |              ^~~~~~~~
```

--- a/Telegram/SourceFiles/boxes/peers/edit_peer_permissions_box.cpp
+++ b/Telegram/SourceFiles/boxes/peers/edit_peer_permissions_box.cpp
@@ -79,7 +79,7 @@ struct NestedRestrictionLabels {
 		second.erase(
 			ranges::remove(
 				second,
-				Flag::CreateTopics,
+				Flag::CreateTopics | Flag(),
 				&RestrictionLabel::flags),
 			end(second));
 	}
@@ -1002,7 +1002,7 @@ std::vector<AdminRightLabel> AdminRightLabels(
 			result.erase(
 				ranges::remove(
 					result,
-					Flag::ManageTopics,
+					Flag::ManageTopics | Flag(),
 					&AdminRightLabel::flags),
 				end(result));
 		}

--- a/Telegram/SourceFiles/history/view/history_view_list_widget.h
+++ b/Telegram/SourceFiles/history/view/history_view_list_widget.h
@@ -355,6 +355,11 @@ protected:
 	int resizeGetHeight(int newWidth) override;
 
 private:
+	using ScrollTopState = ListMemento::ScrollTopState;
+	using PointState = HistoryView::PointState;
+	using CursorState = HistoryView::CursorState;
+	using ChosenReaction = HistoryView::Reactions::ChosenReaction;
+
 	struct MouseState {
 		MouseState();
 		MouseState(
@@ -405,10 +410,6 @@ private:
 		Selecting,
 		Deselecting,
 	};
-	using ScrollTopState = ListMemento::ScrollTopState;
-	using PointState = HistoryView::PointState;
-	using CursorState = HistoryView::CursorState;
-	using ChosenReaction = HistoryView::Reactions::ChosenReaction;
 
 	void onTouchSelect();
 	void onTouchScrollTimer();

